core/modules/global/vectorUtil.h

```diff
@@ -0,0 +1,71 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_VECTORUTIL_H
+#define  LSST_QSERV_VECTORUTIL_H
+ /**
+  * @file
+  *
+  * @brief  Misc. lightweight vector manipulation.
+  *
+  * @author Fabrice Jammes, IN2P3/SLAC
+  */
+
+// System headers
+#include <vector>
+#include <iostream>
+#include <iterator>
+#include <string>
+
+// Third-party headers
+#include "boost/algorithm/string/join.hpp"
+
+// LSST headers
+
+// Qserv headers
+
+namespace lsst {
+namespace qserv {
+
+/**
+ * Print an iterable data structure
+ *
+ * Output format is [a, b, c, ...]
+ * The elements stored in the iterable data structure
+ * must provide an ouput operator.
+ * Examples:
+ * - works fine for std::vector<std::string>,
+ * - doesn't work for std::multimap.
+ *
+ * @param v an iterable data structure
+ * @return a string containing the serialized data structure
+ */
+template<typename Iterable>
+std::string toString(Iterable const& v) {
```

Comment:
Fabrice, anything is possible in C++, if you can define `toString()` for vectors it should also be possible to define some other construct to format vector into a stream :)

My objection to `toString()` is based on couple of of observations:
- most of the time in our code converting complex objects to string is done to push them to streams or to logging system (logging is also usually based of streams)
- methods like toString() are usually implemented using temporary streams.

So if you write code like `cout << toString(vector)` or `LOGF_DEBUG("vector: %1%" % toString(vector))` it is very inefficient because it creates temporary stream and temporary string(s).

To make it more efficient you have to define `operator<<()` which is implemented without using `toString()`. Then you could implement `toString()` based on `operator<<()` but I'd argue that you should avoid it. In case you really need to convert to string there semi-standard tools which already do the same for types that have `operator<<` defined (like `boost::lexical_cast`), but again most of the time you only need `operator<<` as you don't want to mess with strings.

If you want to know how to implement `operator<<` for vector (or any container) here is the sketch of what I would do (there might be simpler ways):

    namespace detail {
        template <typename Cont> struct _ContInserterHelper {
            const Cont& cont;
        };
        template <typename Cont> std::ostream& operator<<(std::ostream& out, const _ContInserterHelper<Cont>& cins) {
            out << "[";
            const Cont& cont = cins.cont;   // this is container itself
            // print container elements with separators
            return out << "]";
        }
    }
    template <typename Cont> detail::_ContInserterHelper<Cont> ContInserter(const Cont& cont) {
        return detail::_ContInserterHelper<Cont>{cont};
    }

And after that you can do:

    std::vector<int> v;
    std::cout << ContInserter(v);

And this has no overhead or any temporary objects created.